File: programmers/코딩테스트 입문 with Java/저주의숫자3.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120871
 */
public class 저주의숫자3 {
    public static void main(String[] args) {
        System.out.println(solution(15));
        System.out.println(solution(40));
    }

    public static int solution(int n) {
        int answer = 0;
        int cnt = 0;
        while (cnt != n) {
            answer++;
            if (answer % 3 == 0 || String.valueOf(answer).contains("3")) continue;
            cnt++;
        }
        return answer;
    }
}

File: programmers/코딩테스트 입문 with Python/저주의 숫자3.py
def solution(n):
    answer = 0
    cnt = 0

    while True:
        if answer % 3 != 0 and str(answer).count('3') == 0:
            cnt += 1
        if cnt == n:
            break
        answer += 1
    return answer


print(solution(15))
print(solution(40))
